app/plugin_api.go

```diff
@@ -317,6 +317,14 @@ func (api *PluginAPI) UpdateUserStatus(userID, status string) (*model.Status, *m
 	return api.app.GetStatus(userID)
 }
 
+func (api *PluginAPI) SetUserStatusTimedDND(userID string, endTime int64) (*model.Status, *model.AppError) {
+	// read-after-write bug which will fail if there are replicas.
+	// it works for now because we have a cache in between.
+	// FIXME: make SetStatusDoNotDisturbTimed return updated status
```

Comment:
Do we want to fix this before merge?

Reply:
this needs fixing in UpdateUserStatus api as well. basically currently no SetStatus* api returns new status so I think it's best to handle in separate pr